packages/siwe/lib/utils.ts

```diff
@@ -42,5 +42,9 @@ export const checkContractWalletSignature = async (
  * an alphanumeric character set.
  */
 export const generateNonce = (): string => {
-  return randomStringForEntropy(96);
+  const nonce = randomStringForEntropy(96);
+  if (nonce.length < 8 || !nonce) {
```

Comment:
Shouldn't the conditions be in the opposite order, to avoid calling `length` on `undefined` or something like that? Not exactly sure about the reason behind `!nonce`

Reply:
indeed